libs/model/src/featurelayer.cpp

```diff
@@ -83,11 +83,127 @@ TileFeatureLayer::TileFeatureLayer(
 
 TileFeatureLayer::~TileFeatureLayer() = default;
 
+/*
+ * Checks if featureIdParts match one of the uniqueIdComposition specifications.
+ * The order of values in KeyValuePairs must be the same as in the composition!
+ */
+bool TileFeatureLayer::validFeatureId(
+    FeatureTypeInfo const& typeId,
+    KeyValuePairs const& featureIdParts,
+    bool excludeTilePrefix) {
+
+    for (auto& candidateComposition : typeId.uniqueIdCompositions_) {
+        auto compositionPart = candidateComposition.begin();
+
+        if (excludeTilePrefix && this->featureIdPrefix().has_value()) {
+
+            // Iterate past the prefix in the unique id composition.
+            auto featureIdPrefixSize = this->featureIdPrefix().value()->size();
+
+            if (featureIdPrefixSize >= candidateComposition.size()) {
+                // TODO check and raise error when the tile prefix is set?
```

Comment:
Not sure about this TODO, wdym?

Reply:
The check was here to avoid a potential runtime error in iteration, but instead it should not be possible to set a tile prefix that's longer (or the same size?) as a composite. This will be also covered by the update.

Reply:
Actually, the new implementation theoretically does not complain if one of the unique id compositions is as long as the tile prefix. It's a weird corner case, so I guess it does not matter... or should I add the check?

Reply:
I think it's fine 👍